torchtitan/metrics.py

```diff
@@ -93,26 +100,55 @@ def build_device_memory_monitor():
         f"{device_type.upper} capacity: {device_memory_monitor.device_name} ({device_memory_monitor.device_index}) "
         f"with {device_memory_monitor.device_capacity_gib:.2f}GiB memory"
     )
-
     return device_memory_monitor
 
 
 class MetricLogger:
-    def __init__(self, log_dir, tag, enable_tb):
+    def __init__(self, log_dir, tag, enable_tb, enable_wandb=False, wandb_config=None):
         self.tag = tag
         self.writer: Optional[SummaryWriter] = None
-        if enable_tb:
+        self.use_wandb = False
+
+        if enable_wandb and WANDB_AVAILABLE:
+            self.use_wandb = True
+            if wandb.run is None:
+                project_name = (
+                    wandb_config.get("project", "torchtitan")
+                    if wandb_config
+                    else "torchtitan"
+                )
+                wandb.init(
+                    project=project_name,
+                    config=wandb_config,
+                    dir=log_dir,
+                )
+            logger.debug("WandB logging enabled")
+        elif enable_tb:
             self.writer = SummaryWriter(log_dir, max_queue=1000)
+            logger.info(f"TensorBoard logging enabled. Logs will be saved at {log_dir}")
+        else:
+            logger.warning("Neither TensorBoard nor WandB logging is enabled.")
 
     def log(self, metrics: Dict[str, Any], step: int):
-        if self.writer is not None:
+        """Log metrics to the configured backend."""
+        if self.use_wandb:
+            wandb_metrics = {
+                (k if self.tag is None else f"{self.tag}/{k}"): v
+                for k, v in metrics.items()
+            }
+            wandb_metrics["step"] = step
+            wandb.log(wandb_metrics)
+        elif self.writer is not None:
```

Comment:
for readability, we can change this line to `elif self.enable_tb` and then `assert self.writer is not None`